src/api/src/main/java/engineering/everest/lhotse/api/rest/annotations/AdminOrAdminOfTargetOrganization.java

```diff
@@ -13,6 +13,6 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Inherited
 @Documented
-@PreAuthorize("hasRole('ADMIN') or (hasRole('ORG_ADMIN') and #requestingUser.organizationId == #organizationId)")
+@PreAuthorize("hasRole('ADMIN') or (hasCustomRole('ORG_ADMIN') and belongsToOrg(#organizationId))")
```

Comment:
`SecurityExpressionRoot#hasRole` is a final method. We can't override it as `CustomMethodSecurityExpression#hasRole`. 

We can have a different method signature but,
we are using the `hasRole` because it works only for keycloak roles checking. We can make any user as super admin by adding `ADMIN` role in keycloak.

We can't use it to validate our app roles like `ORG_ADMIN` and `ORG_USER` because these roles info will be stored in user attributes in keycloak. This was the reason, we have to introduce custom method security expression to handle it.

This issue might be fixed if we move roles management completely to one system(keycloak or our app). Use keycloak roles or use keycloak user attributes but not both.

@sluehr we may have to re-consider our decision and take action again.

Reply:
We should revisit it and document the reason for not following the more obvious approach (if it still makes sense to avoid it).

Reply:
Reason is documented [here](https://github.com/everest-engineering/lhotse/tree/keycloak-integration#custom-method-security-expression-handlers).

Reply:
This was discussed. Consensus is that this is the correct approach, but also that we should consider taking it even further and removing the need for having a local user table. More discussions to come.